specification/index.bs

```diff
@@ -203,3 +283,48 @@ Issue(62): Specify localization handling.
 ## Current behavior of cookie partitioning
 
 # Version number handling
+
+# Algorithms
+
+## Determine the URL for matching a document
+
+To determine the URL to use for matching a document, given the document, `match_origin_as_fallback` and `match_about_blank`:
+
+1. Let |url| be the document's URL.
+1. If the [=scheme=] of |url| is `http`, `https` or `file`:
+    1. Return |url|.
+1. If the [=scheme=] of |url| is `blob`, `data` or `filesystem`, or if |url| is `about:blank` or `about:srcdoc`:
+    1. If `match_origin_as_fallback` is set to `true`:
+        1. If the document's origin is a [=tuple origin=]:
+            1. Let |document-origin| be the serialization of the document's origin.
+            1. If the [=scheme=] of |document-origin| is `http`, `https` or `file`:
+                1. Return |document-origin|.
+            1. Else, return null.
+        *. NOTE: If not a [=tuple origin=], the document’s origin is an [=opaque origin=].
+        1. Let |precursor-origin| be the serialization of the document’s precursor origin, if any.
```

Comment:
```suggestion
        1. Let |precursor-origin| be the serialization of the document’s precursor origin, if any.
            *. TODO: "precursor origin" concept needs to be specified. It is not in the HTML spec at the moment. At least Chrome and Firefox recognize the concept, see e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1715167
```

Reply:
I added this in https://github.com/w3c/webextensions/pull/542/commits/1a40c6569cc81bcad4e3f0170250322f30458d1e.